### PR TITLE
Add three files

### PR DESCRIPTION
**Pull request by MohamedAlkhateb**
Add three files

### CODE REVIEW

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Project.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package dev.omaremara.bugtracker.model;
+
+public enum Project {
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Project.java]: `Project` is not an enum, just a class with a single attribute called `name` of type `String`,

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Project.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    private final String uiName;
+
+    private Project(String uiName) {
+        this.uiName = uiName;
+    }
+
+    public String toString() {
+        return this.uiName;
+    }
+}
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Project.java]: New line at end of file.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import dev.omaremara.bugtracker.model.Project;
+
+public class Report {
+    String title;
+    String description;
+    ReportLevel level;
+    ReportPriority priority;
+    ReportType type;
+    Project project;
+    Assigne assigne;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: `assigne` should be of type `User`.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    public ReportPriority getPriority() {
+        return priority;
+    }
+
+    public ReportType getType() {
+        return type;
+    }
+}
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: New line at end of file.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        return project;
+    }
+
+    public ReportPriority getPriority() {
+        return priority;
+    }
+
+    public ReportType getType() {
+        return type;
+    }
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: Add a `toString` function that returns the name of the user.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/ReportLevel.java [hunk 6]
```diff
@@ -11,6 +11,7 @@
   private ReportLevel(String uiName) { this.uiName = uiName; }
 
   @Override
+
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/ReportLevel.java]: Redundant change.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package dev.omaremara.bugtracker.model;
+
+public class User {
+    String userName;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java]: There is no `userName`, only `email`, `name`, `password`, and `userRole`.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+}
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java:26]: New line at end of file.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        return userName;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getPassword() {
+        return password;
+    }
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java]: Add a `toString` function that returns the name of the user.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
- Add an enum called `UserRole`:
  ```
  enum UserRole {
  TESTER,
  DEVELOPER,
  MANAGER,
  ADMINISTRATOR
  }
  ```
- Add the files to the `classes` file.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java (hunk 3 above)
Comment (MohamedAlkhateb) [reply src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: what do you mean User is class not enum?!

Comment (OmarEmaraDev) [reply src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: There is no type called `assignee`, `assignee` is just a user, so this line should be `User assigne;`

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/UserRole.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package dev.omaremara.bugtracker.model;
+
+public enum UserRole{
+    TESTER("Tester"),
+    DEVELOPER("Developer"),
+    PROJECT_MANGER("Project_Manger"),
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/UserRole.java]: `Project_Manger` should be `Project Manger`, it is a UI name, not a variable.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+public class Report {
+  public String uiName;
+  public String title;
+  public String description;
+  public ReportLevel level;
+  public ReportPriority priority;
+  public ReportType type;
+  public Project project;
+  public User assigne;
+  // function return all dispcription
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: Remove this comment.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
+package dev.omaremara.bugtracker.model;
+
+import dev.omaremara.bugtracker.model.Assigne;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: This class does not exist.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  class Report
+  (String uiName, String title, String description, ReportLevel level,
+   ReportPriority priority, ReportType type, Project project, User assigne) {
+    this.title = title;
+    this.assigne = assigne;
+    this.description = description;
+    this.level = level;
+    this.priority = priority;
+    this.project = project;
+    this.uiName = uiName;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: There is no `uiName` here. There should be an integer `id` attribute.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    this.type = type;
+  }
+
+  @Override
+  public String toString() {
+    return this.uiName;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: This should return the title.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
+package dev.omaremara.bugtracker.model;
+
+public class User {
+  public String email;
+  public String password;
+  public UserRole userRole;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java:9]: Import the UserRole class.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Make sure everything is formatted using clang format.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java (hunk 13 above)
Comment (MohamedAlkhateb) [reply src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: why i want to insert uiName to database to make foriegn key to user name in table user

Comment (OmarEmaraDev) [reply src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/Report.java]: A report doesn't have a "name". The assignee is the attribute that should be a foriegn key to the users table. But a name can't be used as primary key in the users table, so we will either use an id or an email.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 16]
```diff
@@ -0,0 +1,19 @@
+package dev.omaremara.bugtracker.model;
+import dev.omaremara.bugtracker.model.UserRole
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java]: Imports end with a semicolon.

File: src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+  public String email;
+  public String password;
+  public UserRole userRole;
+  public String name;
+
+  class User
+  (String email, String password, UserRole userRole, String name) {
+    this.email = email;
+    this.password = password;
+    this.userRolec = userRole;
```
Comment (OmarEmaraDev) [inline src/dev.omaremara.bugtracker/dev/omaremara/bugtracker/model/User.java]: Typo.

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Did you even try to compile this?
Also, please use clang format.

**Pull request closed without merging**